controllers/storagecluster/external_resources.go

```diff
@@ -230,9 +230,49 @@ func (r *StorageClusterReconciler) newExternalCephObjectStoreInstances(
 // ensureCreated ensures that requested resources for the external cluster
 // being created
 func (obj *ocsExternalResources) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
-	if r.sameExternalSecretData(instance) {
-		return reconcile.Result{}, nil
+
+	if isExternalOCSProvider(instance) {
+
+		externalClusterClient, err := r.newExternalClusterClient(instance)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.externalClusterClient = externalClusterClient
```

Comment:
We do not need `r.externalClusterClient` please remove

Reply:
yes we do else we need to pass it.